### PR TITLE
Add a `pong_delay` option for excessive pings

### PR DESCRIPTION
**Pull request by kitcambridge**
`pong_delay` waits to respond to client pings, to work around clients that enter a ping loop.

### CODE REVIEW

File: autopush/settings.py [hunk 1]
```diff
@@ -51,6 +51,7 @@ def __init__(self,
                  pingConf=None,
                  resolve_hostname=False,
                  min_ping_interval=20,
+                 pong_delay=0,
                  max_data=4096,
```
Comment (jrconlin) [inline autopush/settings.py]: nit, we should make max_data a config option as well. It's a number and we may want to change it.

**Comment by bbangert** [comment]
Needs config example updated. And if you could add min_pong_interval too, I missed that.

File: autopush/settings.py [hunk 2]
```diff
@@ -51,6 +51,7 @@ def __init__(self,
                  pingConf=None,
                  resolve_hostname=False,
                  min_ping_interval=20,
```
Comment (jrconlin) [inline autopush/settings.py]: I thought we were going to not make this hard-coded?

Comment (kitcambridge) [inline autopush/settings.py]: This is the default, no?

Comment (jrconlin) [inline autopush/settings.py]: Yes. I'm a dope. Nevermind.

File: autopush/websocket.py [hunk 3]
```diff
@@ -392,15 +393,24 @@ def finish_notifications(self, notifs):
             self._check_notifications = False
             reactor.callLater(1, self.process_notifications)
 
+    def _send_ping(self):
+        self.last_ping = time.time()
+        self.metrics.increment("updates.client.ping")
```
Comment (bbangert) [inline autopush/websocket.py]: Needs to include tags=self.base_tags.

**Comment by bbangert** [comment]
r+

**Pull request merged**